tokio/src/runtime/blocking/pool.rs

```diff
@@ -220,54 +220,51 @@ impl fmt::Debug for BlockingPool {
 
 impl Spawner {
     pub(crate) fn spawn(&self, task: Task, rt: &Handle) -> Result<(), ()> {
-        let shutdown_tx = {
-            let mut shared = self.inner.shared.lock();
+        let add_thread = |shared: &mut MutexGuard<'_, Shared>, shutdown_tx| {
+            if let Some(shutdown_tx) = shutdown_tx {
+                let id = shared.worker_thread_index;
```

Comment:
You only call this once. Just put it where you use it.

Reply:
The original code use `let  shutdown_tx = {}` may be to consider that the let block needs to be called multiple times in the future?

And the static dispatch closure should have no additional overhead?

Reply:
I'm not worried about overhead. I just think it would be easier to read the code without the closure.

Reply:
I don't think the closure makes code more difficult to read,  it's more clear.

Reply:
But I'll change it if you think it's better without  the closure..